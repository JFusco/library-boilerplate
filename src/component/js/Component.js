'use strict';

import React, { Component, PropTypes } from 'react';
import ComponentTwo from './ComponentTwo';

import '../scss/styles.scss';

class SomeComponent extends Component {
	state = {

	};

	static propTypes = {

	};

	static defaultProps = {

	};

	constructor(props){
		super(props);
	}

	componentWillMount(){
		if (__DEV__){
			console.log('component mounted');
		}
	}

	render(){
		return (
			<div>
				<span>Component</span>

				<ComponentTwo />
			</div>
		);
	}
}

export default SomeComponent;
