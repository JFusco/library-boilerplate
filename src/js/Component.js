'use strict';

import React, { Component, PropTypes } from 'react';
import ComponentTwo from './ComponentTwo';

class SomeComponent extends Component {
	state = {

	};

	static propTypes = {

	};

	static defaultProps = {

	};

	constructor(props){
		super(props);
	}

	componentWillMount(){

	}

	render(){
		return (
			<div>
				<span>Component</span>

				<ComponentTwo />
			</div>
		);
	}
}

export default SomeComponent;
