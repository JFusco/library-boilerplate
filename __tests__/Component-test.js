'use strict';

jest.disableAutomock();

import React from 'react';
import { findDOMNode } from 'react-dom';
import { createRenderer, Simulate, renderIntoDocument } from 'react-addons-test-utils';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Component from '../src/component/js/Component';

describe('Suggestion', () => {
	it('should render', () => {
		const renderer = createRenderer();

		renderer.render(
			<Component />
		);

		const component = renderer.getRenderOutput();

		expect(component.type).toBe('div');
	});
});

describe('Suggestion snapshot', () => {
	it('should render', () => {
		const component = renderer.create(
			<Component />
		).toJSON();

		expect(component).toMatchSnapshot();
	});
});
