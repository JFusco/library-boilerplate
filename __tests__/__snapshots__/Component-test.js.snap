// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestion snapshot should render 1`] = `
<div>
  <span>
    Component
  </span>
  <div>
    Component Two
  </div>
</div>
`;
